### PR TITLE
chore: clean up

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

cleaning up and reusing some shared logics

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by jeroenvervaeke — APPROVED** [review]

File: tools/cli/internal/openapi/openapi.go [hunk 1]
```diff
@@ -28,21 +28,10 @@ type Parser interface {
...
-	Security     openapi3.SecurityRequirements `json:"security,omitempty" yaml:"security,omitempty"`
-	Servers      openapi3.Servers              `json:"servers,omitempty" yaml:"servers,omitempty"`
-	Tags         openapi3.Tags                 `json:"tags,omitempty" yaml:"tags,omitempty"`
-	Info         *openapi3.Info                `json:"info" yaml:"info"`
-	Paths        *openapi3.Paths               `json:"paths" yaml:"paths"`
-	Components   *openapi3.Components          `json:"components,omitempty" yaml:"components,omitempty"`
-	ExternalDocs *openapi3.ExternalDocs        `json:"externalDocs,omitempty" yaml:"externalDocs,omitempty"`
-}
-
-func (o *OasDiff) MergeOpenAPISpecs(paths []string) (*Spec, error) {
```
Comment (andreaangiolillo) [inline tools/cli/internal/openapi/openapi.go:45]: we need this to have the preserve the order of the fields similar to what we have right now in the MMS oas

Comment (blva) [reply tools/cli/internal/openapi/openapi.go:45]: wooo TIL

Comment (andreaangiolillo) [reply tools/cli/internal/openapi/openapi.go:45]: maybe you could add a comment above the struct so that the next time we don't forget about it

Comment (blva) [reply tools/cli/internal/openapi/openapi.go:45]: on it

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**